tough/src/cache.rs

```diff
@@ -68,6 +67,18 @@ impl<'a, T: Transport> Repository<'a, T> {
             "max_timestamp_size argument",
             &metadata_outdir,
         )?;
+
+        for name in self.targets.signed.role_names() {
+            if let Some(filename) = self.delegated_filename(name) {
+                self.cache_file_from_transport(
+                    filename.as_str(),
+                    self.limits.max_targets_size,
```

Comment:
Is this the correct size limit?

```rust
                    self.limits.max_targets_size,
```

Reply:
Good catch. This should be the size in `snapshot.json` for the file if it's present, and if not fall back to max_targets_size.

This also brings to my attention that `fn cache_file_from_transport` should probably just determine the appropriate max size (and hash!) if caching from the transport, although this function may as well serialize the in-memory representations of these roles.

The larger scope of my questions can be punted to an issue.

Reply:
The problem with serializing is that any change could would render a different hash. Fortunately when I was working on caching serde was kind enough to make a small perturbation so that I saw the peril.

Reply:
Oh, right, we put lengths / hashes in the repo metadata. sigh